lib/api/include/irods/genquery2.h

```diff
@@ -0,0 +1,45 @@
+#ifndef IRODS_GENQUERY2_H
+#define IRODS_GENQUERY2_H
+
+/// \file
+
+struct RcComm;
+
+/// The input data type used to invoke #rc_genquery2.
+///
+/// \since 4.3.2
+typedef struct GenQuery2Input // NOLINT(modernize-use-using)
```

Comment:
consider adding a comment explaining the need for `NOLINT(modernize-use-using)`

Reply:
We've never left comments for comments.

Is the NOLINT confusing?

Reply:
Or maybe we should treat some clang-tidy checks as warnings depending on whether we're in C territory. That's may not be what you were getting at, but it's a reasonable thing to do IMO.

There's no `using` directive in C AFAIK so the NOLINT shouldn't be needed for a C header file.

According to https://releases.llvm.org/13.0.1/tools/clang/tools/extra/docs/clang-tidy/index.html, we should be able to change the config applied for files on the fly. From their documentation:
<pre>
Configuration files:
  clang-tidy attempts to read configuration for each source file from a
  .clang-tidy file located in the closest parent directory of the source
  file. If InheritParentConfig is true in a config file, the configuration file
  in the parent directory (if any exists) will be taken and current config file
  will be applied on top of the parent one. If any configuration options have
  a corresponding command-line option, command-line option takes precedence.
  The effective configuration can be inspected using -dump-config:
</pre>

I think this is what we want for directories containing only C header files. Will investigate.

Reply:
I've included an additional clang-tidy file in this PR just for the lib/api/include/irods directory.

If the clang-tidy github action does not reference the genquery2.h file, then I think we're good.